### DISCUSSION TITLE
Access the Users Session in an Edge Function

### DISCUSSION THREAD

**Discussion by TQuantab** [supabase/supabase#20517] [Questions]
hi =)
im currently trying to make a connection with the Microsoft Graph API. In order to do that, i need the Providertoken and Providerrefreshtoken from the Supabase session.

How can i access the Session in my Edge Function (It will run on a Trigger, when a new user gets inserted). Is there any alternative Way, to access the providertokens? I could not find them in the Database.

Thanks for the help =)

**Comment by GaryAustin1** [comment]
When your users signs up or in the provider tokens are in the user object of the session on the client.   They are only there until the session refreshes.  Supabase does not store them. It is up to you to find a method to store them.  I do not know if they are available in the auth trigger function as they are not saved to the database so I would not thing in the insert data for the trigger.
